### PR TITLE
Add more api docs for libflatpak

### PR DESCRIPTION
**Pull request by matthiasclasen**
Bring the library docs much closer to completion.

### CODE REVIEW

File: common/flatpak-transaction.c [hunk 1]
```diff
@@ -508,24 +590,67 @@ flatpak_transaction_operation_get_bundle_path (FlatpakTransactionOperation *self
...
 flatpak_transaction_operation_get_metadata (FlatpakTransactionOperation *self)
 {
   return self->resolved_metakey;
 }
 
+/**
+ * flatpak_transaction_operation_get_old_metadata:
+ * @self: a #FlatpakTransactionOperation
+ *
+ * Gets the metadata that will is applicable before the
```
Comment (alexlarsson) [inline common/flatpak-transaction.c]: This is not right.
old_metadata is the metadata of the current installation, which will be replaced by the one in get_metadata() after the update. You can diff the two to see if we got any new permissions.

File: common/flatpak-transaction.c [hunk 2]
```diff
@@ -872,6 +1055,14 @@ flatpak_transaction_set_disable_prune (FlatpakTransaction *self,
   priv->disable_prune = disable_prune;
 }
 
+/**
+ * flatpak_transaction_set_disable_dependencies:
+ * @self: a #FlatpakTransaction
+ * @disable_dependencies: whether to disable dependencies
+ *
+ * Sets whether the transaction should ignore dependencies when resolving
+ * operations.
```
Comment (alexlarsson) [inline common/flatpak-transaction.c]: Maybe we should mention what dependencies are (i.e. the runtime the app needs).

File: common/flatpak-transaction.c [hunk 3]
```diff
@@ -881,6 +1072,14 @@ flatpak_transaction_set_disable_dependencies (FlatpakTransaction *self,
   priv->disable_deps = disable_dependencies;
 }
 
+/**
+ * flatpak_transaction_set_disable_related:
+ * @self: a #FlatpakTransaction
+ * @disable_related: whether to avoid adding related refs
+ *
+ * Sets whether the transaction should avoid adding related refs
+ * when resolving operations.
```
Comment (alexlarsson) [inline common/flatpak-transaction.c]: Maybe we should mention what related refs are. I.e. extensions suggested by the app such as locales.

File: common/flatpak-transaction.c (hunk 1 above)
Comment (matthiasclasen) [reply common/flatpak-transaction.c]: I was making an educated guess...

File: common/flatpak-transaction.c (hunk 2 above)
Comment (matthiasclasen) [reply common/flatpak-transaction.c]: Yes, I'll add that

File: common/flatpak-transaction.c (hunk 3 above)
Comment (matthiasclasen) [reply common/flatpak-transaction.c]: and that

**Pull request closed without merging**